### PR TITLE
feat(apple-pay-certs): Add `/apple-pay-certificates` endpoints [TA-3300] 

### PR DESCRIPTION
**Pull request by szeying-lee**
# Description
Adds descriptions, resources and endpoints of the `/apple-pay-certificates` endpoints.

Fixes # (issue)
[TA-3300](https://gr4vy.atlassian.net/browse/TA-3300)

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own changes
- [x] I have run `yarn test` to make sure my changes pass all linters
- [x] I have pulled the latest changes from the upstream `main` branch

## Contribution guidelines

For contribution guidelines, styleguide, and other helpful information please
see the `CONTRIBUTING.md` file in the root of this project.

[TA-3300]: https://gr4vy.atlassian.net/browse/TA-3300?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: reference/openapi.yaml [hunk 1]
```diff
@@ -48,6 +48,21 @@ tags:
       guide.
     x-internal: true
 
+  - name: Apple Pay Certificates
+    description: |-
+      Apple Pay payment processing certificates are used by Apple to encrypt Apple
+      Pay tokens. You must register and upload an Apple Pay payment processing
+      certificate if you wish to use Apple Pay with Gr4vy mobile SDKs.
```
Comment (cbetta) [inline reference/openapi.yaml:55]:
```suggestion
      certificate if you wish to use Apple Pay with Gr4vy's mobile SDKs.
```

File: reference/paths/apple_pay_certificates__add-apple-pay-certificate.yaml [hunk 2]
```diff
@@ -0,0 +1,41 @@
+---
+operationId: add-apple-pay-certificate
+summary: New Apple Pay certificate registration
```
Comment (cbetta) [inline reference/paths/apple_pay_certificates__add-apple-pay-certificate.yaml:3]:
```suggestion
summary: Start registering an Apple Pay certificate
```

All endpoints start with a verb.

File: reference/paths/apple_pay_certificates__delete-apple-pay-certificate.yaml [hunk 3]
```diff
@@ -0,0 +1,24 @@
+---
+operationId: delete-apple-pay-certificate
+summary: Delete an Apple Pay certificate
```
Comment (cbetta) [inline reference/paths/apple_pay_certificates__delete-apple-pay-certificate.yaml:3]:
```suggestion
summary: Remove Apple Pay certificate
```

File: reference/resources/apple_pay_certificate.yaml [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+    example: fe26475d-ec3e-4884-9553-f7356683f7f9
+    description: The unique identifier for this Apple Pay certificate record.
+    format: uuid
+
+  status:
+    type: string
+    description: The status of this Apple Pay certificate.
+    example: enabled
+    enum:
+      - enabled
```
Comment (cbetta) [inline reference/resources/apple_pay_certificate.yaml:28]:
```suggestion
      - active
```

Let's change it as agreed

**Review by cbetta — CHANGES_REQUESTED** [review]
Left a few small comments for changes. Rest looks great.

File: reference/paths/apple_pay_certificates__add-apple-pay-certificate.yaml [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+              public_key_hash: null
+              certificate_signing_request: |-
+                -----BEGIN CERTIFICATE REQUEST-----\nMIIBHzCBxwIBADBUMQswCQY
+                DVQQGEwJVSzEQMA4GA1UECAwHRW5nbGFuZDEPMA0G...\n-----END CERTI
+                FICATE REQUEST-----
+              expires_at: null
+              created_at: "2022-12-12T10:53:43+00:00"
+              updated_at: "2022-12-12T10:53:43+00:00"
+
+  "401":
```
Comment (douglaseggleton) [inline reference/paths/apple_pay_certificates__add-apple-pay-certificate.yaml:36]: Do we need to list any of the other error status codes?

File: reference/paths/apple_pay_certificates__list-apple-pay-certificates.yaml [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+      application/json:
+        schema:
+          $ref: "../openapi.yaml#/components/schemas/ApplePayCertificates"
+        examples:
+          A list of Apple Pay certificate records:
+            value:
+              items:
+                - type: apple-pay-certificate
+                  id: fe26475d-ec3e-4884-9553-f7356683f7f9
+                  status: enabled
```
Comment (douglaseggleton) [inline reference/paths/apple_pay_certificates__list-apple-pay-certificates.yaml:25]: Did we change this to active?

File: reference/paths/apple_pay_certificates__add-apple-pay-certificate.yaml (hunk 5 above)
Comment (cbetta) [reply reference/paths/apple_pay_certificates__add-apple-pay-certificate.yaml:36]: Generally we only document the 401s and 403s I think. Don't think 403 is relevant here.

File: reference/openapi.yaml (hunk 1 above)
Comment (szeying-lee) [reply reference/openapi.yaml]: Done: https://github.com/gr4vy/gr4vy-openapi/pull/253/commits/1fb3b7c46bf34637287750c2fd8639535a47be0b

File: reference/resources/apple_pay_certificate.yaml (hunk 4 above)
Comment (szeying-lee) [reply reference/resources/apple_pay_certificate.yaml]: Changed: https://github.com/gr4vy/gr4vy-openapi/pull/253/commits/e697131dd7235e06b35a05e2b3674dd57a6b801c

File: reference/paths/apple_pay_certificates__list-apple-pay-certificates.yaml (hunk 6 above)
Comment (szeying-lee) [reply reference/paths/apple_pay_certificates__list-apple-pay-certificates.yaml]: https://github.com/gr4vy/gr4vy-openapi/pull/253/commits/e697131dd7235e06b35a05e2b3674dd57a6b801c

File: reference/openapi.yaml [hunk 7]
```diff
@@ -48,6 +48,21 @@ tags:
...
+
+      The Apple Pay certificates API can be used to:
+
+      * Start a new Apple Pay certificate registration, providing you with a
+      Certificate Signing Request (CSR).
+      * Update the Apple Pay certificate record with the certificate received from
+      Apple after creating a new payment processing certificate on your Apple
+      Developer console and uploading a CSR.
+      * List all Apple Pay certificates.
```
[the remark this answers is not in the document]
Comment (szeying-lee) [reply reference/openapi.yaml:64]: Are these endpoints be internal? I've kinda written the descriptions for a merchant audience 😅 .
Added the flag here anyway (and to the other new resources): https://github.com/gr4vy/gr4vy-openapi/pull/253/commits/15da8fb43dfc5b07bc5579b711935ac55651c676

[the remark this answers is not in the document]
Comment (cbetta) [reply reference/openapi.yaml:70]: I think we can keep them internal for now until a merchant needs them.

**Review by cbetta — APPROVED** [review]

**Review by douglaseggleton — APPROVED** [review]
Looks good to me - Only question from me is around the need for `x-internal: true`?

**Pull request merged**